### PR TITLE
(#21) Integration tests

### LINKED ISSUE #21: Create loader tests on ES 5
Create tests for loader using Elasticsearch 5.6 so we can verify the ES upgrade doesn't break anything.

- Overall list of documents
- Shape of synonym and category display documents
- queries used by API.

### PR DESCRIPTION
**Pull request by blairl-nih**
- Closes #21

### CODE REVIEW

**Review by bryanpizzillo — COMMENTED** [review]
I have one question. I do have concerns with just "adding tests" that changes any logic -- even if it is better. the change of branchName to ref, and the lowercasing of items. These are changes that I cannot regression test.

File: lib/transformers/category-to-match-transformer.js [hunk 1]
```diff
@@ -61,13 +61,13 @@ class CategoryToMatchTransformer extends AbstractRecordTransformer {
         const incMatches = data.includeSynonyms
                             .map(syn => ({
                                 ...(this.extractMatchFromSyn(data, false, syn)),
-                                tokenCount: lookup[syn.name]
+                                tokenCount: lookup[syn.name.toLowerCase()]
```
[the remark this answers is not in the document]
Comment (blairl-nih) [reply lib/transformers/category-to-match-transformer.js:64]: The logic change is a separate commit for #23.  The backstory is that when I was writing the test to verify the shape of synonyms (integration-tests/features/smoketest.feature, lines 45-69), I discovered there were a small number of synonyms being saved with no `tokencount`.

I believe that lacking a `tokencount` means this queries from the API's `ESBestBetsMatchService` will be unable to find those synonyms.

```csharp
           var matchQuery = new NumericRangeQuery { Field = "tokencount", LessThanOrEqualTo = matchedTokenCount } &&
                             new TermQuery { Field = "is_exact", Value = 0 } &&
                             new TermQuery { Field = "language", Value = lang } &&
                             new MatchQuery { Field = "synonym", Query = cleanedTerm, MinimumShouldMatch = matchedTokenCount };

            if (searchTokenCount == matchedTokenCount)
            {
                //Add in exact match query too
                matchQuery = matchQuery ||
                             new TermQuery { Field = "tokencount", Value = matchedTokenCount } &&
                             new TermQuery { Field = "is_exact", Value = 1 } &&
                             new TermQuery { Field = "language", Value = lang } &&
                             new MatchQuery { Field = "synonym", Query = cleanedTerm, MinimumShouldMatch = matchedTokenCount };
            }
```

Looking at the `smoketest.feature` file, I see I left a couple (no-longer accurate) comments in, including a commented-out `@ignore` tag that was originally meant to disable that test.  I'll remove those comments and re-push in a moment.

[the remark this answers is not in the document]
Comment (blairl-nih) [reply lib/transformers/category-to-match-transformer.js:64]: I should also point out, the lower-cased values are not saved to Elasticsearch.  They're only used within the logic for tracking the tokencount values for a given synonym.

**Comment by blairlearn** [comment]
Updated to remove the normalization commit and modify tests to accept missing `numtoken`.

File: lib/transformers/category-to-match-transformer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply lib/transformers/category-to-match-transformer.js]: I am resolving this comment as this bug fix was removed from the initial tests.

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**